### PR TITLE
Cli nested sed ml fix

### LINKED ISSUE #291: VCell CLI can not handle nested sed-documents
### Background
- Omex archives can contain nested SED-ML files; we have declared to biosimulators that we can handle these cases, and have in the past.
- The biosimulators test suite checks for failures regarding nested SED-ML files.

### Problem:
- We get a cascade of exceptions attempting to run the synthetic archives run by biosimulators test suite.

### PR DESCRIPTION
**Pull request by CodeByDrescher**
fixes #291

### CODE REVIEW

**Review by jcschaff — CHANGES_REQUESTED** [review]

File: vcell-cli/src/main/java/org/vcell/cli/run/ExecuteImpl.java [hunk 1]
```diff
@@ -223,11 +226,13 @@ public static void singleExecOmex(File inputFile, File rootOutputDir, boolean bK
 
                 // Converting pseudo SED-ML to biomodel
                 logger.info("Creating Biomodel from pseudo SED-ML");
-                sedmlFromPseudo = Libsedml.readDocument(sedmlPathwith2dand3d).getSedMLModel();
+                sedmlFromPython = Libsedml.readDocument(sedmlPathwith2dand3d).getSedMLModel();
```
[the remark this answers is not in the document]
Comment (CodeByDrescher) [reply vcell-cli/src/main/java/org/vcell/cli/run/ExecuteImpl.java:229]: I compromised and put it near the top of the relevant scope. If this is not preferable, I'll do inline definition...I just don't like doing that in general

File: vcell-cli/src/main/java/org/vcell/cli/run/ExecuteImpl.java [hunk 2]
```diff
@@ -394,6 +399,28 @@ public static void singleExecOmex(File inputFile, File rootOutputDir, boolean bK
...
     }
 
+    /**
+     * In its current state, the sed-ml generated with python is missing two important fields;
+     * this function fixes that.
+     */
+    private static SedML repairSedML(SedML brokenSedML, String[] tokenizedPath){
+        for (int i = 0; i < tokenizedPath.length; i++){
+            if (tokenizedPath[i].startsWith("vcell_temp")){
+
```
[the remark this answers is not in the document]
Comment (CodeByDrescher) [reply vcell-cli/src/main/java/org/vcell/cli/run/ExecuteImpl.java:409]: Ill go with option 1

File: vcell-core/src/main/java/org/jlibsedml/execution/ArchiveModelResolver.java [hunk 3]
```diff
@@ -15,8 +15,10 @@ public String getModelXMLFor(URI modelURI) {
         String rc=null;
         List<IModelContent> children = ac.getModelFiles();
         for (IModelContent imc: children) {
-            if(imc.getName().equals(modelURI.toString())){
-                rc= imc.getContents();
+            String modelElementStr = imc.getName(), modelStr = modelURI.toString();
+            //if(imc.getName().endsWith(modelURI.toString())){
```
[the remark this answers is not in the document]
Comment (CodeByDrescher) [reply vcell-core/src/main/java/org/jlibsedml/execution/ArchiveModelResolver.java:19]: I tend to try and keep like types on the same line to reduce the number of lines in a scope, but this scope is small enough anyways.

File: vcell-core/src/main/java/org/vcell/sedml/SEDMLImporter.java [hunk 4]
```diff
@@ -132,6 +133,7 @@ private void initialize() throws FileNotFoundException, XMLException {
 		if(externalDocInfo.getFile().getPath().toLowerCase().endsWith("sedx") || externalDocInfo.getFile().getPath().toLowerCase().endsWith("omex")) {
 			ac = Libsedml.readSEDMLArchive(new FileInputStream(externalDocInfo.getFile().getPath()));
 		}
+		List<SEDMLDocument> lsd = ac.getSedmlDocuments(); //DEBUG: DELETE BEFORE MERGE
```
[the remark this answers is not in the document]
Comment (CodeByDrescher) [reply vcell-core/src/main/java/org/vcell/sedml/SEDMLImporter.java:136]: yes...thank you...
see I knew that would come in handy!

File: vcell-cli/src/main/java/org/vcell/cli/run/ExecuteImpl.java [hunk 5]
```diff
@@ -128,28 +130,28 @@ public static void singleExecOmex(File inputFile, File rootOutputDir, boolean bK
         OmexHandler omexHandler = null;
         List<String> sedmlLocations;
         List<Output> outputs;
-        SedML sedml, sedmlFromPseudo;
+        SedML sedml, sedmlFromPython;
```
[the remark this answers is not in the document]
Comment (jcschaff) [reply vcell-cli/src/main/java/org/vcell/cli/run/ExecuteImpl.java]: fixed

File: vcell-cli/src/main/java/org/vcell/cli/run/ExecuteImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcschaff) [reply vcell-cli/src/main/java/org/vcell/cli/run/ExecuteImpl.java:229]: good compromise ... I prefer inline personally, but don't feel strongly.

File: vcell-cli/src/main/java/org/vcell/cli/run/ExecuteImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcschaff) [reply vcell-cli/src/main/java/org/vcell/cli/run/ExecuteImpl.java]: nice.

File: vcell-core/src/main/java/org/jlibsedml/execution/ArchiveModelResolver.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (jcschaff) [reply vcell-core/src/main/java/org/jlibsedml/execution/ArchiveModelResolver.java]: good - that style reminds me of C code (a bit slower to read for me), but it is fine for me.

File: vcell-core/src/main/java/org/vcell/sedml/SEDMLImporter.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (jcschaff) [reply vcell-core/src/main/java/org/vcell/sedml/SEDMLImporter.java]: cool

**Review by jcschaff — APPROVED** [review]
Approved

**Pull request merged**